src/tls/settings.rs

```diff
@@ -317,6 +333,20 @@ impl TlsConfig {
         }
     }
 
+    fn parse_alpn_protocols(&self) -> Result<Option<Vec<u8>>> {
```

Comment:
Might be helpful to add a comment here describing the format required:

The input must be in ALPN "wire format". It consists of a sequence of supported protocol names prefixed by their byte length.